### DISCUSSION TITLE
Link a proforma e ddt

### DISCUSSION THREAD

**Discussion by dmigliacci** [fattureincloud/api#306] [Q&A]
Buongiorno,
Stiamo positivamente utilizzando le api per generare proforme e ddt.

Alla creazione, salviamo i link pdf della proforma e del ddt collegato.

Abbiamo però notato, che quando tale proforma viene trasformata in fattura, seguendo il normale processo di vendita, i link sopra indicati non sono più validi e non è possibile risalire al ddt necessario alla spedizione.

Ho notato che proforma e ddt vengono spostati dal sistema nel cestino, che andrebbe bene, ma il link al documento pdf cambia url.

Da: https://compute.fattureincloud.it/**doc/cad**/xxxx.pdf

A: https://compute.fattureincloud.it/**doc/bi1/cad/bi1**/xxxx.pdf

Potete cambiare questo comportamento di rinominare le url dei pdf?

Grazie,
Davide

**Comment by emajo** [comment]
Ciao @dmigliacci, il link viene cambiato perché appunto spostato nel cestino successivamente verrà eliminato, quindi il mio consiglio è di non basarsi su quell'url, una soluzione sarebbe quella di non eliminare la proforma quando viene trasformato il documento (settando `keep_copy = true`) in modo che l'url non cambi, oppure se non si vuole mantenere il documento forse la migliore opzione è quella di scaricare una copia del pdf una volta creata la proforma.

Grazie,
Emanuele.

**Reply by valmoz** [reply]
Buongiorno @dmigliacci 
è riuscito a risolvere?

Grazie
Mauro

**Reply by dmigliacci** [reply]
Ciao e scusate il ritardo.
Grazie per la spiegazione, la capisco, ma sto ancora pensando ad una soluzione.

Al momento (e anche nel futuro) la trasformazione da Proforma a Fattura sarà fatta direttamente su fattureincloud.
Ho però verificato che anche web è possibile mantenere la proforma originale tramite la spunta.
Potremmo iniziare a lasciarla spuntata, tenendo le proforme originali e poi buttarle noi manualmente ogni tot giorni.

Pensando ad un'altra soluzione, faccio un'altra proposta: non potreste rispondere 404 alla richiesta di un pdf non valido, o meglio non più trovato?
Perchè intercettando quello potrei gestirlo in maniera diversa, mentre con una 200 ok non riconosco la differenza.

Grazie dell'attenzione,
Davide
